packages/controls/src/controls/shape/v2/shape-v2.test.ts

```diff
@@ -34,388 +30,152 @@ function deserializer(record: DeserializedRecord): ControlDefinition {
-              checkbox: Checkbox(),
-            },
+  describe('deserialized `Group` with `shape-v2::v1` data type', () => {
+    const shapeV2 = new GroupDefinition(
+      {
+        label: 'Inline shape',
+        preferredLayout: Group.Layout.Inline,
+        props: {
+          checkbox: Checkbox({
+            label: 'Checkbox',
           }),
+          color: Color({ label: 'Color', defaultValue: 'red' }),
         },
-      })
-
-      const [serialized, _] = shape.serialize()
-      expect(serialized).toMatchSnapshot()
-
-      const deserialized = ShapeV2Definition.deserialize(
-        deserializeRecord(serialized),
-        deserializer,
-      )
-
-      expect(deserialized).toEqual(shape)
-    })
-  })
+      },
+      ShapeV2Definition.v1DataType,
+    )
 
-  describe('introspection', () => {
-    test('Shape data is correctly delegated to subcontrols', () => {
-      const shape = ShapeV2({
-        type: {
-          color: Color({ defaultValue: 'red' }),
-          link: Link(),
-          image: Image(),
+    const data: DataType<GroupDefinition> = {
+      [ControlDataTypeKey]: ShapeV2Definition.v1DataType,
+      value: {
+        checkbox: {
+          [ControlDataTypeKey]: 'checkbox::v1',
+          value: true,
         },
-      })
-
-      const shapeData = {
         color: {
-          swatchId: 'swatch-id',
+          [ControlDataTypeKey]: 'color::v1',
+          swatchId: '[swatchId]',
           alpha: 1,
         },
-        link: {
-          type: 'OPEN_PAGE' as const,
-          payload: {
-            pageId: 'page-id',
-            openInNewTab: false,
-          },
-        },
-        image: 'file-id',
-      }
-
-      const swatchIds = shape.introspect(shapeData, Targets.Swatch)
-      expect(swatchIds).toEqual(['swatch-id'])
-
-      const pageIds = shape.introspect(shapeData, Targets.Page)
-      expect(pageIds).toEqual(['page-id'])
-
-      const fileIds = shape.introspect(shapeData, Targets.File)
-      expect(fileIds).toEqual(['file-id'])
-    })
-
-    test('ShapeV2 versioned data is correctly delegated to subcontrols', () => {
-      const shape = ShapeV2({
-        type: {
-          color: Color({ defaultValue: 'red' }),
-          link: Link(),
-          image: Image(),
-        },
-      })
-
-      const shapeData = {
-        [ControlDataTypeKey]: 'shape-v2::v1' as const,
-        value: {
-          color: {
-            swatchId: 'swatch-id',
-            alpha: 1,
-          },
-          link: {
-            type: 'OPEN_PAGE' as const,
-            payload: {
-              pageId: 'page-id',
-              openInNewTab: false,
-            },
-          },
-          image: 'file-id',
-        },
-      }
-
-      const swatchIds = shape.introspect(shapeData, Targets.Swatch)
-      expect(swatchIds).toEqual(['swatch-id'])
+      },
+    }
 
-      const pageIds = shape.introspect(shapeData, Targets.Page)
-      expect(pageIds).toEqual(['page-id'])
+    test('safeParse', () => {
```

Comment:
Here and below, test that "shapeV2" operations on data preserve its data signature.